[CUBRIDMAN-4] english manual for CBRD-23608 (TDE) (#249)
CUBRID/cubrid-manual: en/admin/admin_utils.rst

    cubrid addvoldb -S -n cubriddb_DATA01 --db-volume-size=512M cubriddb
    cubrid addvoldb -S -p temp -n cubriddb_TEMP01 --db-volume-size=512M cubriddb

.. note:: **Creating a database using an existing key file**

>     When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database, first copy the key file with the name <database-name>_keys. Afterward, specify the directory path of the copied key file by the **ted_keys_file_path** system parameter, and create a database through the **createdb** utility. For more information on the TDE key file, see :ref:`tde-file-based-key`.

Reviewer: ```suggestion
    When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database, first copy the key file with the name <database-name>_keys. Afterward, specify the directory path of the copied key file by the system parameter (**ted_keys_file_path**) and create a database through the **createdb** utility. For more information on the TDE key file, see :ref:`tde-file-based-key`.
```

- comment: sentence maybe too long or hard to read, consider rephrase it.

Author: How about this?
- I removed parentheses you suggetsed for consistency with others
```suggestion
    When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database:
#.  Copy the key file with the name <database-name>_keys.
#.  Specify the directory path of the copied key file by the system parameter **tde_keys_file_path**.
#.  Create a database by using the **createdb** utility.

    For more information on the TDE key file, see :ref:`tde-file-based-key`.
```

Reviewer: good!

>     When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database, first copy the key file with the name <database-name>_keys. Afterward, specify the directory path of the copied key file by the **ted_keys_file_path** system parameter, and create a database through the **createdb** utility. For more information on the TDE key file, see :ref:`tde-file-based-key`.

Reviewer 2: typo miss
```suggestion
    When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database, first copy the key file with the name <database-name>_keys. Afterward, specify the directory path of the copied key file by the **tde_keys_file_path** system parameter, and create a database through the **createdb** utility. For more information on the TDE key file, see :ref:`tde-file-based-key`.
```

Author: Thanks

>     When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database, first copy the key file with the name <database-name>_keys. Afterward, specify the directory path of the copied key file by the **tde_keys_file_path** system parameter, and create a database through the **createdb** utility. For more information on the TDE key file, see :ref:`tde-file-based-key`.

Author: ```suggestion
    When the database is created, a key file is created together by default. If you want to use an existing key file when creating a database:
#.  Copy the key file with the name <database-name>_keys.
#.  Specify the directory path of the copied key file by the system parameter **tde_keys_file_path**.
#.  Create a database by using the **createdb** utility.

    For more information on the TDE key file, see :ref:`tde-file-based-key`.
```

        cubrid paramdump -C demodb

.. _tde-utility:

tde
---

> The **cubrid tde** utility is used to manage the TDE encryption of the database, and can only be executed by the **DBA** user. With **cubrid tde** utility, you can change the key set on the database, and add a new key and remove a key in the key file stably. Also, you can inquire about the keys added to the key file and the key set on the database. For more information, see :ref:`tde`.

Reviewer: ```suggestion
The **cubrid tde** utility is used to manage the TDE encryption of the database and can only be executed by the **DBA** user. With **cubrid tde** utility, you can change the key set on the database, add a new key, or remove a key in the key file stably. Also, you can inquire about the keys added to the key file and the key set on the database. For more information, see :ref:`tde`.
```

-  ', ' in here seems to be unnecessary

- if the functions are not coherent, changing 'and' to 'or' is recommended

::

    cubrid tde <operation> [option] database_name

*   **cubrid**: An integrated utility for the CUBRID service and database management

*   **tde**: A utility that manages TDE encryption applied to the database

> *   *operation*: There are four types of operation: key addition, key deletion, key change, and showing keys' information. One operation must be given and they are exclusive.

Reviewer: ```suggestion
*   *operation*: There are four types of operation: key addition, key deletion, key change, and showing keys' information. One operation must be set, and they are exclusive.
```

> *   *database_name*: The name of the database on which TDE administration operations are to be performed

Reviewer: ```suggestion
*   *database_name*: The name of the database on which TDE administration operations to be performed
```

- 'are' and 'to be' both indicate passive form, consider deleting one of them

.. program:: tde

.. option:: -s, --show-keys

    This option displays information about the keys set on the database and keys in the key file (_keys).  ::

        $ cubrid tde -s testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        The current key set on testdb:
       Key Index: 2
        Created on Fri Nov 27 11:14:54 2020
        Set     on Fri Nov 27 11:15:30 2020

        Keys Information:
        Key Index: 0 created on Fri Nov 27 11:11:27 2020
        Key Index: 1 created on Fri Nov 27 11:14:47 2020
        Key Index: 2 created on Fri Nov 27 11:14:54 2020
        Key Index: 3 created on Fri Nov 27 11:14:55 2020

        The number of keys: 4

    **The current key** section shows the information on the key set on the current database, which displays the index for the key in the key file, the creation time, and the set time. The set key can be identified by the key index and creation time, and a key change plan can be established with the set time.

    **Keys Information** section shows the keys added and being managed in the key file, through which the key indexes and creation time of them is checked.

.. option:: -n, --generate-new-key

>     This option is used to add a new key to the key file (up to 128). If successful, the index of the added key is displayed, and this index is used to identify the key when changing or removing the key later. Information of added keys can be checked by \\-\\-show-keys. ::

Reviewer: ```suggestion
    This option is used to add a new key to the key file (up to 128). If it is successful, the index of the added key is displayed, and this index is used to identify the key when changing or removing the key later. Information of added keys can be checked by \\-\\-show-keys. ::
```

        $ cubrid tde -n testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        SUCCESS: A new key has been generated - key index: 1 created on Tue Dec  1 11:30:47 2020

.. option:: -d, --delete-key=KEY_INDEX

    This option is used to remove a key specified by index from the key file. The key currently set on the database cannot be removed. :: 

        $ cubrid tde -d 1 testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        SUCCESS: The key (index: 1) has been deleted

.. option:: -c, --change-key=KEY_INDEX

>     This option is used to change the key set on the database to another key existing in the key file. When changing, both the previously set key and the newly  key to be set must exist.  ::

Reviewer: ```suggestion
    This option is used to change the key set on the database to another key existing in the key file. When changing, both the previously set key and the new key to be set must exist.  ::
```

        $ cubrid tde -c 2 testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        Trying to change the key from the key (index: 0) to the key (index: 2)..
        SUCCESS: The key has been changed from the key (index: 0) to the key (index: 2)

>     To change the key set on the database, a user must first create a key to be set by the \\-\\-generate-new-key option. The user can create a new key to change, or create multiple keys in advance, and change the key according to theis own security plans.

Reviewer: ```suggestion
    To change the key set on the database, a user must first create a key to be set by the \\-\\-generate-new-key option. The user can create a new key to change, or create multiple keys in advance, and change the key according to their own security plans.
```

        cubrid paramdump -C demodb

.. _tde-utility:

tde
---

The **cubrid tde** utility is used to manage the TDE encryption of the database and can only be executed by the **DBA** user. With **cubrid tde** utility, you can change the key set on the database, add a new key, or remove a key in the key file stably. Also, you can inquire about the keys added to the key file and the key set on the database. For more information, see :ref:`tde`.

::

    cubrid tde <operation> [option] database_name

> *   **cubrid**: An integrated utility for the CUBRID service and database management

Reviewer 3: It must end with a period.

Author: Oh, some of the utilites end with a period, and others are not. I think you're right.
```suggestion
*   **cubrid**: An integrated utility for the CUBRID service and database management.
```

> *   **tde**: A utility that manages TDE encryption applied to the database

Reviewer 3: It must end with a period.

Author: ```suggestion
*   **tde**: A utility that manages TDE encryption applied to the database.
```

*   *operation*: There are four types of operation: key addition, key deletion, key change, and showing keys' information. One operation must be set, and they are exclusive.

> *   *database_name*: The name of the database on which TDE administration operations to be performed

Reviewer 3: It must end with a period.

Author: ```suggestion
*   *database_name*: The name of the database on which TDE administration operations to be performed.
```

The following table shows <operation> available with the cubrid tde utility.

.. program:: tde

.. option:: -s, --show-keys

    This option displays information about the keys set on the database and keys in the key file (_keys).  ::

        $ cubrid tde -s testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        The current key set on testdb:

>        Key Index: 2

Reviewer 3: indentation

Author: ```suggestion
        Key Index: 2
```

        $ cubrid tde -n testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        SUCCESS: A new key has been generated - key index: 1 created on Tue Dec  1 11:30:47 2020

.. option:: -d, --delete-key=KEY_INDEX

    This option is used to remove a key specified by index from the key file. The key currently set on the database cannot be removed. :: 

        $ cubrid tde -d 1 testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        SUCCESS: The key (index: 1) has been deleted

.. option:: -c, --change-key=KEY_INDEX

    This option is used to change the key set on the database to another key existing in the key file. When changing, both the previously set key and the new key to be set must exist.  ::

        $ cubrid tde -c 2 testdb
        Key File: /home/usr/CUBRID/databases/testdb/testdb_keys

        Trying to change the key from the key (index: 0) to the key (index: 2)..
        SUCCESS: The key has been changed from the key (index: 0) to the key (index: 2)

>     To change the key set on the database, a user must first create a key to be set by the \\-\\-generate-new-key option. The user can create a new key to change, or create multiple keys in advance, and change the key according to their own security plans.

Reviewer 2: ```suggestion
    To change the key set on the database, a user must first create a key to be set by the \\-\\-generate-new-key option. The user can create a new key to change, or create multiple keys in advance and change the key according to their own security plans.
```
or 

```suggestion
    To change the key set on the database, a user must first create a key to be set by the \\-\\-generate-new-key option. The user can create a new key to change, or create multiple keys in advance for changing the key according to their own security plans.
```

I think comma makes its meaning confusing
